src/block/client/componolit-interfaces-block-client.ads

```diff
@@ -54,26 +52,111 @@ package Componolit.Interfaces.Block.Client with
+
+   --  Get request start block
+   --
+   --  @param R  Request
+   --  @return   First block id to be handled by this request
+   function Start (R : Request) return Id with
+      Pre => Status (R) /= Raw;
+
+   --  Get request length
+   --
+   --  @param R  Request
+   --  @return   Number of consecutive blocks handled by this request
+   function Length (R : Request) return Count with
+      Pre => Status (R) /= Raw;
+
+   --  Get request identifier
+   --
+   --  @param R  Request
+   --  @return   Unique identifier of the request
+   function Identifier (R : Request) return Request_Id with
+      Pre => Status (R) /= Raw;
+
+   --  Allocate request
+   --
+   --  This procedure allocates a request on the platform. This means setting the intended request range
+   --  and identifier but also allocating the backing store for the block data. This procedure can fail,
+   --  e.g. if not enough memory is available to allocate the block data store. In the case of success
+   --  the request status will change from Raw to Allocated. In case of a failure it will stay Raw.
+   --
+   --  @param C       Client session instance
+   --  @param R       Request to allocate
+   --  @param Kind    Request type
+   --  @param Start   First block to be handled by the request
+   --  @param Length  Number of consecutive blocks to be handled
+   --  @param Ident   Unique identifier for this request, chosen by the application
+   procedure Allocate_Request (C      : in out Client_Session;
+                               R      : in out Request;
+                               Kind   :        Request_Kind;
+                               Start  :        Id;
+                               Length :        Count;
+                               Ident  :        Request_Id) with
+      Pre => Initialized (C) and then Status (R) = Raw;
+
+   --  Check if the request handle is valid
+   --
+   --  @param H  Request handle
+   --  @return   True if handle is valid
+   function Valid (H : Request_Handle) return Boolean;
+
+   --  Get the request identifier linked with the capability
+   --
+   --  @param H  Request handle
+   --  @return   Identifier of the request linked to this capability
+   function Identifier (H : Request_Handle) return Request_Id with
+      Pre => Valid (H);
+
+   --  Check the response queue for updates
+   --
+   --  Reads the first element from the response queue and saves its meta data into the request capability.
+   --  The capability is required to update a request with the according metadata.
+   --
+   --  @param C  Client session instance
+   --  @param H  Platform handle that indicates the request status change
+   procedure Update_Response_Queue (C : in out Client_Session;
+                                    H :    out Request_Handle) with
+      Pre => Initialized (C);
+
+   --  Update request according to request handle
+   --
+   --  Takes a request handle and updates the request according to the platform state
```

Comment:
For a someone new to the interface, it's not apparent what gets updated. We should clarify that in the comment and check for the other subprograms whether they either have an obvious purpose or add a rationale to the comment.